services/dataAccessLayer.js

```diff
@@ -37,9 +46,25 @@ const retreiveFilteredUsers = async (query) => {
   return users;
 };
 
+const retrieveMembers = async (query) => {
+  const allUsers = await members.fetchUsers(query);
+  allUsers.forEach((element) => {
+    removeSensitiveInfo(element);
+  });
+  return allUsers;
+};
+
+const retrieveUsersWithRole = async (role) => {
+  const users = await members.fetchUsersWithRole(role);
+  users.forEach((element) => {
+    removeSensitiveInfo(element);
+  });
+  return users;
+};
+
 const removeSensitiveInfo = function (obj) {
   for (let i = 0; i < USER_SENSITIVE_DATA.length; i++) {
-    if (USER_SENSITIVE_DATA[i] in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, USER_SENSITIVE_DATA[i])) {
```

Comment:
I think we don't need to access hasOwnProperty from prototype, it should be directly accessible